### ISSUE TITLE
cloned dashboards share variable state

### ISSUE THREAD

**Issue by genehynson** [influxdata/ui#3287]
It appears to be a bug that when you clone a dashboard with variables, changing the value of a variable in the cloned dashboard changes it in the original one as well. 

Uploading Screen Recording 2021-11-19 at 10.35.33 AM.mov…

.

**Comment by Palakp41** [comment]
I don't think this issue is only for the Cloning dashboards. I created a new dashboard `A` which uses the same variable `xyz` as an existing dashboard `B`. Then changed the value of `xyz` on dashboard `A` which also changes the value in existing dashboard `B`.

**Comment by Palakp41** [comment]
I tried implementing a UI fix for this by adding a `selectedVariable` state for the specific dashboard user is viewing but it will go back to the global variable default when we refresh this page. So, I think this needs to be a tiny epic. We need a way to store the variable and the value that was selected for a particular dashboard in the backend, then we can update the store and manage this state in the UI.
